File: src/main/java/shenefelt/libraryManager/Book.java
package shenefelt.libraryManager;

public class Book
{

    private String title;
    private String author;
    private String genre;
    private final int BOOK_ID = Library.IDManager.generateId();
    private boolean isSeries;

    public Book(String bTitle, String bAuthor, String bGenre, boolean inSeries)
    {
        title = bTitle;
        author = bAuthor;
        genre = bGenre;
        isSeries = inSeries;
    }

    public Book()
    {
        title = author = " ";
        isSeries = false;
    }

    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }

    public String getAuthor() { return author; }
    public void setAuthor(String author) { this.author = author; }

    public String getGenre() { return genre; }
    public void setGenre(String genre) { this.genre = genre; }

    public int getBook_id() { return BOOK_ID; }

    public boolean aSeries() { return isSeries; }
    public void setSeries (boolean series) { isSeries = series; }


    public String display()
    {
    }
}

File: src/main/java/shenefelt/libraryManager/kotlin/Book.kt
package shenefelt.libraryManager.kotlin

class Book(private var title: String = "",
           private var author: String = "",
           private var genre: String = "",
           private var price: Double = 0.00,
           private var series: Boolean = false,
           private var totalPages: Int = 0,
           )
{
    private var seriesList: MutableList<Book>? = null

    init
    {
        println("New book created")
    }


    fun getTitle(): String
    {
        return title;
    }

    fun getAuthor(): String
    {
        return author;
    }

    fun getGenre(): String
    {
        return genre;
    }

    fun getPrice(): Double
    {
        return price;
    }

    fun getSeries(): Boolean
    {
        return series;
    }

    fun getTotalPages(): Int
    {
        return totalPages;
    }

    fun setTitle(newTitle: String)
    {
        title = newTitle;
    }

    fun setAuthor(newAuthor: String)
    {
        author = newAuthor;
    }

    fun setGenre(newGenre: String)
    {
        genre = newGenre;
    }

    fun setPrice(newPrice: Double)
    {
        price = newPrice;
    }

    fun setSeries(newSeries: Boolean)
    {
        if(newSeries) collectSeriesInfo()
        series = newSeries;
    }

    fun setTotalPages(newTotalPages: Int)
    {
        totalPages = newTotalPages;
    }

    fun displayBook()
    {
        println("Title: $title")
        println("Author: $author")
        println("Genre: $genre")
        println("Price: $price")
        println("Series: $series")
    }

    private fun collectSeriesInfo()
    {
        if(seriesList == null) seriesList = mutableListOf();

        println("Book Title: ")

    }
}